File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (const auto& c : s)
        {
            switch(c)
            {
                case ')':
                if (st.size() == 0 || st.top() != '(') return false;
                st.pop();
                continue;
                case '}':
                if (st.size() == 0 ||st.top() != '{') return false;
                st.pop();   
                continue;
                case ']':
                if (st.size() == 0 ||st.top() != '[') return false;
                st.pop();
                continue;
                default:
                st.push(c);
            }
        }
        return st.size() == 0;
    }
};

File: Data Structures & Algorithms/minimum-stack/submission-1.py
class MinStack:
    s = []
    m = []

    def __init__(self):
        self.s = []
        self.m = []
        

    def push(self, val: int) -> None:
        self.s.append(val)
        if len(self.m) == 0 or self.m[-1] >= val: 
            self.m.append(val)

    def pop(self) -> None:
        v = self.s.pop()
        if self.m[-1] == v:
            self.m.pop()

    def top(self) -> int:
        return self.s[-1]

    def getMin(self) -> int:
        return self.m[-1]
